application/common/models/User.php

```diff
@@ -442,11 +480,15 @@ public function getMfaFields()
     {
         return [
             'prompt'  => $this->isPromptForMfa() ? 'yes' : 'no',
-            'nag'     => (! $this->isPromptForMfa() && $this->isTimeToNagForMfa()) ? 'yes' : 'no',
+            'add'     => $this->getNagState() == self::NAG_ADD_MFA ? 'yes' : 'no',
```

Comment:
Are we going to have logic here in ID Broker to ensure that, if `prompt` is `yes` then no nags will happen? It seems like it would be good to avoid doing both, if we can, but I don't see anything here that would prevent them both from happening at the same time (the way we did in the old code).

Reply:
I looked at the consumer code you sent, and it was clear that `prompt` was the priority. If `prompt` is `yes`, then `nag` is ignored. Didn't make sense to duplicate the logic here and prevent the consumer from implementing the logic as best fits a future scenario. I even considered making `getNagState` obsolete by simply providing all appropriate nags and leaving the priority up to the consumer, though I didn't go that far since it wasn't the original plan.

Reply:
Okay, great. Glad you did it this way, then.